backend/typescript/services/implementations/userService.ts

```diff
@@ -0,0 +1,240 @@
+import * as firebaseAdmin from "firebase-admin";
+
+import IUserService from "../interfaces/userService";
+import MgUser, { User } from "../../models/user.mgmodel";
+import { CreateUserDTO, UpdateUserDTO, UserDTO } from "../../types";
+import Logger from "../../utilities/logger";
+
+class UserService implements IUserService {
+  /* eslint-disable class-methods-use-this */
```

Comment:
We're not using any class properties so all the methods here can be made `static`, but I want to stick with our design of using the interface.